crates/pallet-humanode-session/src/lib.rs

```diff
@@ -72,6 +76,73 @@ pub mod pallet {
     #[pallet::storage]
     pub type CurrentSessionIndex<T: Config> = StorageValue<_, SessionIndex, OptionQuery>;
 
+    /// A set of all banned accounts that can't be validators in the network.
+    #[pallet::storage]
+    pub type BannedAccounts<T: Config> =
+        StorageValue<_, BoundedBTreeSet<T::AccountId, T::MaxBannedAccounts>, ValueQuery>;
```

Comment:
But why?..

Use `StorageMap`?

Reply:
Either do that, or drop the `is_banned` API and replace with something that operates on the entire value - as `is_banned` will pay the price of load the whole thing while only using part of it; we'd be better of with loading the whole thing and running the filtering over with maintaining the same access, instead of dropping and re-acquiring the ref (there is storage cache, but we'd still pay for the deconding/enoding as I understand).

Or, just use map if only the access to individual keys is anticipated (which might be less optimal here).

Reply:
Good point, but I would prefer to user `StorageValue` with `BTreeSet` as it should be more optimal here. But, sure, we should avoid dropping and re-acquiring the ref, reimplemented https://github.com/humanode-network/humanode/pull/1249/commits/cfeb9204f445c07105bc7f397db3b0c4b5bb45c5, `is_banned` API has been dropped.